accounts/abi/abi_test.go

```diff
@@ -745,3 +745,35 @@ func TestABI_MethodById(t *testing.T) {
 		t.Errorf("Expected error, nil is short to decode data")
 	}
 }
+
+func TestABI_EventById(t *testing.T) {
+	const abiJSON = `[
+		{"type":"event","name":"received","anonymous":false,"inputs":[
+			{"indexed":false,"name":"sender","type":"address"},
+			{"indexed":false,"name":"amount","type":"uint256"},
+			{"indexed":false,"name":"memo","type":"bytes"}
+			]
+		}]`
```

Comment:
so yeah please test with 0 event, 1 event and more than one event

Reply:
as I said in my previous comment, a topic has a single event and an event has a single topic (hash is unique).

My test already includes a case for
* event can be found
* event cannot be found